mesonbuild/mtest.py

```diff
@@ -99,6 +99,29 @@ def uniwidth(s: str) -> int:
         result += UNIWIDTH_MAPPING[w]
     return result
 
+def test_slice(arg: str) -> T.Tuple[int, int]:
+    values = arg.split('/')
+    if len(values) != 2:
+        raise argparse.ArgumentTypeError("value does not conform to format 'SLICE/NUM_SLICES'")
+
+    try:
+        nrslices = int(values[1])
+    except ValueError:
+        raise argparse.ArgumentTypeError("NUM_SLICES is not an integer")
+    if nrslices <= 0:
+        raise argparse.ArgumentTypeError("NUM_SLICES is not a positive integer")
+
+    try:
+        subslice = int(values[0])
+    except ValueError:
+        raise argparse.ArgumentTypeError("SLICE is not an integer")
+    if subslice <= 0:
+        raise argparse.ArgumentTypeError("SLICE is not a positive integer")
+    if subslice > nrslices:
+        raise argparse.ArgumentTypeError("SLICE exceeds NUM_SLICES")
```

Comment:
Silly nitpicky nit: using double quotes where the rest of the commit (and file) uses single quotes, even with prose.

Reply:
Fixed, as well, except for the one instance where we use single quotes in the string itself.